File: HeartForCharity/HeartForCharity.Model/Requests/VolunteerJobUpdateRequest.cs
using HeartForCharity.Model.Enums;
using System;

namespace HeartForCharity.Model.Requests
{
    public class VolunteerJobUpdateRequest
    {
        public int? CategoryId { get; set; }
        public int? AddressId { get; set; }
        public string Title { get; set; } = null!;
        public string? Description { get; set; }
        public string? Requirements { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public bool IsRemote { get; set; } = false;
        public int PositionsAvailable { get; set; }
        public VolunteerJobStatus Status { get; set; }
    }
}

File: HeartForCharity/UI/heartforcharity_desktop/lib/model/requests/volunteer_job_update_request.dart
class VolunteerJobUpdateRequest {
  final int? categoryId;
  final int? addressId;
  final String title;
  final String? description;
  final List<int> skillIds;
  final DateTime? startDate;
  final DateTime? endDate;
  final bool isRemote;
  final int positionsAvailable;
  final String? status;

  VolunteerJobUpdateRequest({
    this.categoryId,
    this.addressId,
    required this.title,
    this.description,
    this.skillIds = const [],
    this.startDate,
    this.endDate,
    this.isRemote = false,
    required this.positionsAvailable,
    this.status,
  });

  Map<String, dynamic> toJson() => {
        'categoryId': categoryId,
        'addressId': addressId,
        'title': title,
        'description': description,
        'skillIds': skillIds,
        'startDate': startDate?.toUtc().toIso8601String(),
        'endDate': endDate?.toUtc().toIso8601String(),
        'isRemote': isRemote,
        'positionsAvailable': positionsAvailable,
        'status': status,
      };
}
